src/Text/Pandoc/App.hs

```diff
@@ -127,6 +127,41 @@ convertWithOpts scriptingEngine opts = do
                Left e -> E.throwIO $ PandocShouldNeverHappenError $ T.pack e
           | otherwise -> writeFnBinary outputFile bs
 
+isolateBigNote :: ([Inline] -> Inline) -> [Inline] -> [Inline]
+isolateBigNote constructor xs
+  | any isBigNote xs = let (before, after) = break isBigNote xs
+              in case after of
+                   (noteInline:rest) -> constructor before : noteInline : isolateBigNote constructor rest
+                   _ -> error "break should always return a non-empty 'after' when 'any isNote xs' is True."
+  | otherwise = [constructor xs]
+  where
+    isBigNote :: Inline -> Bool
+    isBigNote (Note blocks) = countParas blocks > 1
+      where
+        countParas :: [Block] -> Int
+        countParas = length . filter isPara
+
+        isPara :: Block -> Bool
+        isPara (Para _) = True
+        isPara _        = False
+    isBigNote _ = False
+
+liftBigNotes :: [Inline] -> [Inline]
+liftBigNotes [] = []
+liftBigNotes (x:xs) =
+    case x of
+        Emph inner    -> isolateBigNote Emph (liftBigNotes inner) ++ liftBigNotes xs
+        Strong inner  -> isolateBigNote Strong (liftBigNotes inner) ++ liftBigNotes xs
+        _             -> x : liftBigNotes xs
+
```

Comment:
I think this is unnecessary.  If you define
```
liftBigNotes :: Inline -> Inline
liftBigNotes (Emph inner) = ...
etc.
```
then you can lift this `Inline -> Inline` transformation to a `Pandoc -> Pandoc` transformation using `walk`.
```
walk liftBigNotes
```
can be made to have the type Pandoc -> Pandoc, and everything should work (of course, needs testing).

Reply:
Thank you for the feedback.

I'm not sure how to translate this `liftBigNotes :: [Inline] -> [Inline]` to `liftBigNotes :: Inline -> Inline`.
I feel the result values will need to have `[Inline]` type. See the following wrong code.


```haskell
liftBigNotes :: Inline -> Inline
liftBigNotes (Emph inner)    = isolateBigNote Emph (liftBigNotes inner) ++ liftBigNotes xs
liftBigNotes (Strong inner)  = isolateBigNote Strong (liftBigNotes inner) ++ liftBigNotes xs
liftBigNotes others          = [others]
```
This code is wrong because 
1. `liftBigNotes`'s output has the `Inline` type, while the actual function body outputs the `[Inline]` type
2. `liftBigNotes inner`’s inner has the `[Inline]`  type while `liftBigNotes` expects the `Inline` type

Can you elaborate a little more on how to modify this function?

Reply:
Sorry, right.  What about the following, with type `[Inline] -> [Inline]`?
``` hs
liftBigNotes :: [Inline] -> [Inline]
liftBigNotes (Emph inner : xs)    = isolateBigNote Emph (liftBigNotes inner) ++ liftBigNotes xs
liftBigNotes (Strong inner : xs)  = isolateBigNote Strong (liftBigNotes inner) ++ liftBigNotes xs
liftBigNotes (x : xs)          = x : liftBigNotes xs
liftBigNotes [] = []
```
I believe that `walk liftBigNotes` will lift this to a `Pandoc -> Pandoc` transformation.

Reply:
I worry a bit about performance implications, but we can test that.

Reply:
Function is updated. But I might need to address the performance implications later.